lib/http/response.rb

```diff
@@ -92,13 +92,17 @@ def flush
     #   (not an integer, e.g. empty string or string with non-digits).
     # @return [Integer] otherwise
     def content_length
+      # http://greenbytes.de/tech/webdav/rfc7230.html#rfc.section.3.3.3
+      # Clause 3: "If a message is received with both a Transfer-Encoding
+      # and a Content-Length header field, the Transfer-Encoding overrides the Content-Length.        
+      return nil if headers[Headers::TRANSFER_ENCODING]
```

Comment:
also, I have just pushed small improvement to `Headers` class, so this line can benefit of using it:

``` ruby
return nil if @headers.include?(Headers::TRANSFER_ENCODING)
```

`Headers#include?` is more efficient (does not allocates new objects in comparison to `#[]`)